### PR TITLE
[FE-3395] Cropper

### PR DESCRIPTION
**Pull request by xixe**

### CODE REVIEW

File: src/modules/form/components/UploadFileField/components/CropperBlock/CropperBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      cropper.scaleY(newVal);
+      setFlipV(newVal);
+    }
+  };
+
+  const handleReset = useCallback(() => {
+    cropper.reset();
+  }, [cropper]);
+
+  const renderCropperButtons = (
```
Comment (Antroll) [inline src/modules/form/components/UploadFileField/components/CropperBlock/CropperBlock.tsx:75]: `render` - its for function naming
```suggestion
  const renderedCropperButtons = (
```

File: src/modules/form/components/UploadFileField/components/CropperBlock/CropperBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            classes.cropperTransformBtnRestore,
+          )}
+        >
+          <RestoreIcon />
+        </IconButton>
+      </Tooltip>
+
+      <Tooltip title={t('cropper.rotate-left')} arrow>
+        <IconButton
+          onClick={() => handleRotate(-90)}
```
Comment (Antroll) [inline src/modules/form/components/UploadFileField/components/CropperBlock/CropperBlock.tsx:91]: You need to define arrow function. Also please check the rest of the similar use places.

File: src/modules/form/components/UploadFileField/components/ImagePreview/ImagePreview.tsx [hunk 3]
```diff
@@ -1,13 +1,45 @@
...
   fitView: boolean;
+  cropper: boolean;
+  cropHandle: any;
+  cropperAspect: number;
+  CropperPreviewComponent?: JSX.Element;
+  needShowCropper: boolean;
 }
 
-export const ImagePreview = ({ image, fitView }: IImagePreviewProps) => {
+const renderCropper = (
```
Comment (Antroll) [inline src/modules/form/components/UploadFileField/components/ImagePreview/ImagePreview.tsx:16]: Why just not component?

**Review by Antroll — CHANGES_REQUESTED** [review]

File: src/modules/form/components/UploadFileField/UploadFileField.tsx [hunk 4]
```diff
@@ -55,94 +44,177 @@ const renderInitialBlock = (
...
+  const FileComponent = FilePreviewComponent ?? FilePreview;
+  return <FileComponent fileName={fileName} fileSize={fileSize} />;
 };
 
 const renderImagePreview = (
   ImagePreviewComponent: any,
-  image: string,
-  fitView: boolean,
+  image: string | undefined,
+  fitView: boolean | undefined,
```
[the remark this answers is not in the document]
Comment (xixe) [reply src/modules/form/components/UploadFileField/UploadFileField.tsx]: Done

File: src/modules/form/components/UploadFileField/components/CropperBlock/CropperBlock.tsx (hunk 1 above)
Comment (xixe) [reply src/modules/form/components/UploadFileField/components/CropperBlock/CropperBlock.tsx]: Ok, fixed

File: src/modules/form/components/UploadFileField/components/CropperBlock/CropperBlock.tsx (hunk 2 above)
Comment (xixe) [reply src/modules/form/components/UploadFileField/components/CropperBlock/CropperBlock.tsx]: Ok

**Review by Antroll — APPROVED** [review]

**Pull request merged**